sql/core/src/test/resources/sql-tests/inputs/typeCoercion/native/binaryComparison.sql

```diff
@@ -0,0 +1,287 @@
+--
+--   Licensed to the Apache Software Foundation (ASF) under one or more
+--   contributor license agreements.  See the NOTICE file distributed with
+--   this work for additional information regarding copyright ownership.
+--   The ASF licenses this file to You under the Apache License, Version 2.0
+--   (the "License"); you may not use this file except in compliance with
+--   the License.  You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+--   Unless required by applicable law or agreed to in writing, software
+--   distributed under the License is distributed on an "AS IS" BASIS,
+--   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+--   See the License for the specific language governing permissions and
+--   limitations under the License.
+--
+
+-- Binary Comparison
+
+CREATE TEMPORARY VIEW t AS SELECT 1;
+
+SELECT cast(1 as binary) = '1' FROM t;
```

Comment:
Seems binary comparison without [<=>](https://github.com/apache/spark/blob/ced6ccf0d6f362e299f270ed2a474f2e14f845da/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/predicates.scala#L594).

Reply:
yea `<=>` is missed. Do Derby tests also miss it?

Reply:
might be OK as `<=>` should have same type coercion rule as `=`

Reply:
`===` is consistent with `<=>`

I think we can skip it. Then, we can run the whole suite to Hive with minor changes.